### PR TITLE
add initial implementation for Rgroup Decomposition

### PR DESCRIPTION
**Pull request by AlexanderSavelyev**

### CODE REVIEW

**Comment by bp-kelley** [comment]
It appears that this code doesn't actually compile RGroupDecomp.cpp, when added, the file itself doesn't compile errors of the kind:

```
rdkit/Code/GraphMol/ChemTransforms/RGroupDecomp.cpp:36:34: error: 
      type 'std::pair<int, int>' does not provide a subscript operator
            if(matches[mIdx][idx][0] != matches[mIdx][idx][1])
```

Perhaps this pull request is missing some files?

**Comment by AlexanderSavelyev** [comment]
PR was updated by fixing compilation errors

File: Code/GraphMol/ChemTransforms/CMakeLists.txt [hunk 1]
```diff
@@ -15,5 +15,8 @@ rdkit_test(testChemTransforms testChemTransforms.cpp
            LINK_LIBRARIES ChemTransforms FileParsers SmilesParse SubstructMatch
            GraphMol RDGeometryLib RDGeneral ${RDKit_THREAD_LIBS})
 
+rdkit_test(testRGroupDecomp testRGroupDecomp.cpp
```
Comment (bp-kelley) [inline Code/GraphMol/ChemTransforms/CMakeLists.txt]: testRGroupDecomp was not added to the repository

**Comment by bp-kelley** [comment]
Nope, kill it.

**Pull request closed without merging**